src/main/scala/com/fulcrumgenomics/vcf/filtration/EndRepairArtifactLikelihoodFilter.scala

```diff
@@ -0,0 +1,212 @@
+  val FilterDescription = "Variant is likely an artifact caused by end-repair and A-base addition."
+
+  override val VcfInfoLines: Traversable[VCFInfoHeaderLine]     = Seq(vcfInfoLine(InfoKey, InfoDescription))
+  override val VcfFilterLines: Traversable[VCFFilterHeaderLine] = Seq(vcfFilterLine(FilterKey, FilterDescription))
+
+  /** Only applies to het SNVs where the alt allele is A or T. */
+  override def appliesTo(gt: Genotype): Boolean = {
+    gt.isHet &&
+      !gt.isHetNonRef &&
+      gt.getAlleles.forall(_.length() == 1) &&
+      gt.getAlleles.filter(_.isNonReference).exists(a => a.getBaseString == "A" || a.getBaseString == "T")
+  }
+
+  /** Calculates the p-value and returns it in the map of annotations. */
+  override def annotations(pileup: Pileup[PileupEntry], gt: Genotype): Map[String,Any] = {
+    if (!appliesTo(gt)) {
+      Map.empty
+    }
+    else {
+      val refAllele = SequenceUtil.upperCase(gt.getAlleles.filter(_.isReference).map(_.getBases()(0)).next())
+      val altAllele = SequenceUtil.upperCase(gt.getAlleles.filter(_.isNonReference).map(_.getBases()(0)).next())
+      annotations(pileup, refAllele, altAllele)
+    }
+  }
+
+  /**
+    * Workhorse method that does the majority of the calculation of the p-value.
+    */
+  private[filtration] def annotations(pileup: Pileup[PileupEntry], refAllele: Byte, altAllele: Byte) = {
+    // First we need to count the artifact congruent and non-congruent reference bases
+    var (refGood, refBad, altGood, altBad) = (0, 0, 0, 0)
+    pileup.baseIterator.filter(_.base == refAllele).foreach { rec =>
+      if (isArtifactCongruent(refAllele, altAllele, rec, pileup.pos)) refBad += 1
+      else refGood += 1
+    }
+
+    val lnFractionCongruent = LnProb.toLogProbability(refBad / (refGood + refBad).toDouble)
+    val lnFractionIncongruent = LnProb.toLogProbability(refGood / (refGood + refBad).toDouble)
+    var llMutation = 1.0
+    var llArtifact = 1.0
+
+    // Now we can iterate over the alt bases and determine their likelihood under the two models
+    pileup.baseIterator.filter(_.base == altAllele).foreach { rec =>
+      val congruent = isArtifactCongruent(refAllele, altAllele, rec, pileup.pos)
+      val logProbError = (rec.qual / -10.0) * NumericTypes.LnTen // Convert Q into ln(pError)
+      val logProbRight = LnProb.not(logProbError)
+
+      if (congruent) {
+        altBad += 1
+        llArtifact += logProbRight
+        llMutation += LnProb.or(LnProb.and(logProbRight, lnFractionCongruent), LnProb.and(logProbError, lnFractionIncongruent))
+      }
+      else {
+        altGood += 1
+        llArtifact += logProbError
+        llMutation += LnProb.or(LnProb.and(logProbRight, lnFractionIncongruent), LnProb.and(logProbError, lnFractionCongruent))
+      }
+    }
+
+    // Apply a prior based on the MAF
+    val maf = (altGood + altBad) / (altGood + altBad + refGood + refBad).toDouble
+    val priorMutation = min(pow(maf * 2, 2), 1)
```

Comment:
Don't you want `2 * maf * (1-maf)`? I.e. the prior probability that one of the two alleles is alt?